rust/main/agents/scraper/src/db/message.rs

```diff
@@ -31,64 +32,54 @@ pub struct StorableMessage<'a> {
 }
 
 impl ScraperDb {
-    /// Get the dispatched message associated with a nonce.
+    /// Get the delivered message associated with a sequence.
     #[instrument(skip(self))]
-    pub async fn retrieve_message_by_nonce(
+    pub async fn retrieve_delivered_message_by_sequence(
         &self,
-        origin_domain: u32,
-        origin_mailbox: &H256,
-        nonce: u32,
-    ) -> Result<Option<HyperlaneMessage>> {
-        #[derive(Copy, Clone, Debug, EnumIter, DeriveColumn)]
-        enum QueryAs {
-            Nonce,
-        }
-        if let Some(message) = message::Entity::find()
-            .filter(message::Column::Origin.eq(origin_domain))
-            .filter(message::Column::OriginMailbox.eq(address_to_bytes(origin_mailbox)))
-            .filter(message::Column::Nonce.eq(nonce))
+        destination_domain: u32,
+        destination_mailbox: &H256,
+        sequence: u32,
+    ) -> Result<Option<Delivery>> {
+        if let Some(delivery) = delivered_message::Entity::find()
+            .filter(delivered_message::Column::Domain.eq(destination_domain))
```

Comment:
is there any meaningful benefit to selecting just the column we care about? i assume it's pretty much negligible as there isn't much data in each row

Reply:
I am not sure I am following. We are filtering delivered messages by three columns: domain, mailbox and sequence, but we select the whole row. We choose to use only message id because it is the only data we return from this method.